### DISCUSSION TITLE
SQLite Plugin

### DISCUSSION THREAD

**Discussion by muuvmuuv** [tauri-apps/tauri#2527] [Ideas]
A SQLite plugin for Tauri that is bundled with the app or any other SQLite Rust alternatives. 

It does not need all features from beginning. We would need it to store persistent encrypted data that a user collects while using our app in offline mode, so that he can later upload the data to our server.

**Comment by nothingismagick** [comment]
We currently provide a plugin for Stronghold, which is an encrypted-at-rest key:value store as well as a hardened vault for protecting and using high-value secrets. See: 

https://github.com/tauri-apps/tauri-plugin-stronghold

**Reply by muuvmuuv** [reply]
That looks good but we would need a SQL like syntax in our case. Especially because we are programming a hybrid app with capacitor for which we would add a intergration for. There is already a lot of progress integrating Tauri into the Capacitor ecosystem but it lacks some features and plugins for full web, mobile, desktop support.

**Comment by nothingismagick** [comment]
Well, you should be glad to know that there will be a Mobile version of Tauri in 2021.

**Reply by Torbatti** [reply]
2023 still no mobile version :'(

**Reply by FabianLars** [reply]
Since then we learned to be a bit more careful with promises and ETAs but either way, we have an alpha version available and working to get it to beta state asap :)
https://next--tauri.netlify.app/next/mobile/ (also look through our blog posts)
